shell/platform/common/cpp/public/flutter_texture_registrar.h

```diff
@@ -49,11 +62,22 @@ typedef const FlutterDesktopPixelBuffer* (
     *FlutterDesktopPixelBufferTextureCallback)(size_t width,
                                                size_t height,
                                                void* user_data);
+typedef const FlutterDesktopGpuBuffer* (
+    *FlutterDesktopGpuBufferTextureCallback)(size_t width,
+                                               size_t height,
+                                               void* user_data);
+
+typedef void (*FlutterDesktopDestructionCallback)(void* user_data);
 
 // An object used to configure pixel buffer textures.
 typedef struct {
   // The callback used by the engine to copy the pixel buffer object.
   FlutterDesktopPixelBufferTextureCallback callback;
+
+  FlutterDesktopGpuBufferTextureCallback gbCallback;
+
+  FlutterDesktopDestructionCallback destructionCallback;
+
```

Comment:
How about creating a new type like `FlutterDesktopGpuBufferTextureConfig` rather than sharing the existing `FlutterDesktopPixelBufferTextureConfig`? For example,

```cpp
// ...description
typedef const FlutterDesktopGpuBuffer* (
    *FlutterDesktopGpuBufferTextureCallback)(size_t width,
                                            size_t height,
                                            void* user_data);

// ...description
typedef void (*FlutterDesktopGpuBufferCollectCallback)(void* user_data);

// An object used to configure GPU buffer textures.
typedef struct {
  // The callback invoked to obtain the GPU buffer object. (Is this a correct description?)
  FlutterDesktopGpuBufferTextureCallback callback; // (Any better name for this callback?)
  // The callback invoked to release the GPU buffer object.
  FlutterDesktopGpuBufferCollectCallback collect_callback;
  // Opaque data that will get passed to the provided |callback|.
  void* user_data;
} FlutterDesktopGpuBufferTextureConfig;

typedef struct {
  FlutterDesktopTextureType type;
  union {
    FlutterDesktopPixelBufferTextureConfig pixel_buffer_config;
    FlutterDesktopGpuBufferTextureConfig gpu_buffer_config;
  };
} FlutterDesktopTextureInfo;
```

Reply:
Good suggestion，I will implement it.

Reply:
Oh... I think this way is more simple and easy to read. :+1: 

By the way, making more changes to original part, I wonder if this change can be accepted by...  :worried: